### DISCUSSION TITLE
Custom CSS for print format?

### DISCUSSION THREAD

**Discussion by Jerome-thinking** [frappe/print_designer#194] [General]
When creating a new template with the print designer we still would like to add custom styles. Usually we could just place css classes in the "custom css" block in the print format form of the according template. However, with the print designer these css classes seem to be stripped from the final output. At least we could not get custom css to be applied. What is the recommended approach?

**Comment by maharshivpatel** [comment]
@Jerome-thinking can you please share some cases where you would want to have custom css classes.

**Comment by Jerome-thinking** [comment]
Maybe the approach is already wrong:
From the former print formats we were used to set custom css classes. Now when using the print designer, this seems obsolete. When preparing an example I realized clicking on any item and then adjusting the styles works well.

However, just for the record:
We used to assemble custom components using macros. E.g.
```jinja
{% macro formatted_content() %}
<div class="formatted-content">
    div with class ".formatted-content"
</div>
<div style="color:blueviolet;">
    div with inline style (color:blueviolet;)
</div>
{% endmacro %}
```
The css classes were defined in the print format form for the template.

The print designer seems to override the print format css (maybe on save). Just add some css, go to the print designer and play around a bit. Revisit the print format form after a while: your custom css will be gone (at least this happens here).

Given the power of the print designer we could replace our macros by using the builder. But still not sure...

**Reply by maharshivpatel** [reply]
you can define macros [here](https://github.com/frappe/print_designer/pull/115)  and use like [this](https://github.com/frappe/print_designer/pull/99#issue-2031972593) avoid unless absolutely necessary

**Reply by Jerome-thinking** [reply]
But still does not solve the issue with how to apply custom css style to them, doesn´t it?

**Reply by marcramser** [reply]
@Jerome-thinking  did you find a way to do this?

@maharshivpatel  I have the use case that I need would like to load a font that is not selectable in the "default" languages. Any idea on how i can do that?
